internal/patroni/config.go

```diff
@@ -280,8 +280,10 @@ func DynamicConfiguration(
 	for i := range pgHBAs.Mandatory {
 		hba[i] = pgHBAs.Mandatory[i].String()
 	}
-	if section, ok := postgresql["pg_hba"].([]string); ok {
-		hba = append(hba, section...)
+	if _, ok := postgresql["pg_hba"].([]interface{}); ok {
+		for _, section := range postgresql["pg_hba"].([]interface{}) {
+			hba = append(hba, section.(string))
```

Comment:
🔧 This could panic when the input contains not-string.

I'm not sure what we should do if the user gives `pg_hba: ["str1", 5, "str2"]`. The following ignores the values that are not-string:

```go
	if section, ok := postgresql["pg_hba"].([]interface{}); ok {
		for i := range section {
			if value, ok := section[i].(string); ok {
				hba = append(hba, value)
			}
		}
	}
```

Reply:
🤔 Alternatively, we could stop caring about the type of the values:

```go
	hba := make([]interface{}, len(pgHBAs.Mandatory))
	for i := range pgHBAs.Mandatory {
		hba[i] = pgHBAs.Mandatory[i].String()
	}
	if section, ok := postgresql["pg_hba"].([]interface{}); ok {
		hba = append(hba, section...)
	}
```

Reply:
Its starting to feel like we should define a proper structure here for capturing this configuration (e.g. so that we can add proper validation for fields like this), rather than capturing the raw input :thinking:

Reply:
Agreed. I want to avoid baking Patroni structures into the CRD, when possible.

Postgres parameters and HBA aren't Patroni-specific features. We should elevate them out of this area as a separate concern.

Reply:
I updated the logic to simply ignore anything that isn't a string.  I honestly don't know what sort for rule someone would add that would not be a string outside of a mis-configuration (e.g. the format essentially forces a string), and considering a bad `pg_hba.cong` configuration can prevent PG from even starting, this doesn't seem like such a bad thing.  This also aligns with what this would likely look like if defined elsewhere in the spec (array of strings) - which I agree should be elevated outside of this section.